src/main/java/com/ericsson/ei/handlers/StatusHandler.java

```diff
@@ -0,0 +1,222 @@
+        statusData.setEventRepositoryStatus(eventRepositoryStatus);
+
+        Status eiffelIntelligenceStatus = getEiffelIntelligenceStatus();
+        statusData.setEiffelIntelligenceStatus(eiffelIntelligenceStatus);
+    }
+
+    /**
+     * Returns the status of the existing mongoDB connection.
+     *
+     * @return status
+     */
+    private Status getMongoDBStatus() {
+        Status status;
+
+        try {
+            MongoDatabase database = mongoDBHandler.getMongoClient().getDatabase("admin");
+            Document serverStatus = database.runCommand(new Document("serverStatus", 1));
+            Map connections = (Map) serverStatus.get("connections");
+            status = evaluateAvailableConnections(connections);
+        } catch (Exception e) {
+            status = Status.UNAVAILABLE;
+        }
+
+        LOGGER.debug("MongoDB status is {}", status);
+        return status;
+    }
+
+    /**
+     * Determines whether the number of available connections exceeds the required available
+     * connections and returns a status based on the result.
+     *
+     * @param connections
+     * @return status
+     */
+    private Status evaluateAvailableConnections(Map<String, Integer> connections) {
+        Status status;
+        int availableConnections = connections.get("available");
+        int current = connections.get("current");
+
+        if (availableConnections >= REQUIRED_AVAILABLE_CONNECTIONS) {
+            status = Status.AVAILABLE;
+        } else {
+            status = Status.UNAVAILABLE;
+
+            LOGGER.error("Available connections in MongoDB is less than expected.\n"
+                    + "Available connections: '{}' used connections: '{}' expected available connections: '{}'",
+                    availableConnections, current, REQUIRED_AVAILABLE_CONNECTIONS);
+        }
+
+        return status;
+    }
+
+    /**
+     * Returns the status of the existing rabbitMQ connection.
+     *
+     * @return
+     */
+    private Status getRabbitMQStatus() {
+        Status status;
+
+        boolean isActive = rmqHandler.getContainer().isActive();
+        boolean isRunning = rmqHandler.getContainer().isRunning();
+        LOGGER.debug("RabbitMQ is running '{}' is active '{}'.", isRunning, isActive);
+
+        if (isActive && isRunning) {
+            status = Status.AVAILABLE;
+        } else {
+            status = Status.UNAVAILABLE;
+        }
+
+        LOGGER.debug("RabbitMQ status is {}", status);
+        return status;
+    }
+
+    /**
+     * Returns the status of the EventRepository service.
+     *
+     * @return
+     */
+    private Status getEventRepositoryStatus() {
+        Status status;
+        if (StringUtils.isEmpty(erUrl)) {
+            status = Status.DISABLED;
+        } else {
+            // /HealtCheck or /Status endpoint must be available in the service to check.
+            // TODO: HTTPRequest should be made to the ER.
+            status = Status.UNKNOWN;
+        }
+
```

Comment:
I think ER status check can be removed and implemented after ER has been updated/implemented with a healthcheck entrypoint. So create separate issue for ER healtcheck and a separate issue for EI to use and check ER new healthcheck entrypoint.

Reply:
ER check has been commented and will not be performed, keeping the code until it may be reimplemented, commented code id usually not good, but this will most likely be implemented soon.

Reply:
Ok. I think a healthcheck should be implemented in every Eiffel component that has a Web/Rest interface with same input request and with same JSON data response and same response HTTP Error codes. Book a meeting and discuss how a healthcheck entrypoint should work, what inputs and expected outputs from a healthcheck entrypoint.

Reply:
I agree.